src/signal_map.rs

```diff
@@ -258,6 +311,55 @@ mod mutable_btree_map {
         fn signal_map_cloned(&mut self) -> MutableSignalMap<K, V> {
             let (sender, receiver) = mpsc::unbounded();
 
+            if !self.values.is_empty() {
+                sender.unbounded_send(MapDiff::Replace {
+                    entries: Self::entries_cloned(&self.values),
+                }).unwrap();
+            }
+
+            self.senders.push(sender);
+
+            MutableSignalMap {
+                receiver
+            }
+        }
+    }
+
+    impl<K: Copy + Ord, V: Copy> MutableBTreeState<K, V> {
+        fn entries(values: &BTreeMap<K, V>) -> Vec<(K, V)> {
+            values.into_iter().map(|(k, v)| {
+                (*k, *v)
+            }).collect()
+        }
+
+        fn replace(&mut self, values: BTreeMap<K, V>) {
+            let entries = self.change(|| Self::entries(&values));
+
+            self.values = values;
+
+            self.notify_clone(entries, |entries| MapDiff::Replace { entries });
+        }
+
+        fn insert(&mut self, key: K, value: V) -> Option<V> {
+            let x = self.change(|| (key, value));
+
+            if let Some(value) = self.values.insert(key, value) {
+                if x.is_some() {
```

Comment:
I don't like this, since it's not using the value. So instead you should get rid of this branch completely, and also get rid of the `change` call.

They're not needed because the `notify` method already does nothing if there are no receivers.